Dockerfile

```diff
@@ -18,19 +18,22 @@ RUN make bins
 ##### Temporal server #####
 FROM ${BASE_SERVER_IMAGE} AS temporal-server
 WORKDIR /etc/temporal
+
 ENV TEMPORAL_HOME /etc/temporal
 ENV SERVICES "history:matching:frontend:worker"
 EXPOSE 6933 6934 6935 6939 7233 7234 7235 7239
-ENTRYPOINT ["./entrypoint.sh"]
 
 COPY config/dynamicconfig /etc/temporal/config/dynamicconfig
 COPY docker/config_template.yaml /etc/temporal/config/config_template.yaml
 COPY docker/entrypoint.sh /etc/temporal/entrypoint.sh
 COPY docker/start-temporal.sh /etc/temporal/start-temporal.sh
-
 COPY --from=temporal-builder /temporal/tctl /usr/local/bin
 COPY --from=temporal-builder /temporal/temporal-server /usr/local/bin
 
+RUN chown -R temporal:temporal /etc/temporal
+USER temporal
+
+ENTRYPOINT ["./entrypoint.sh"]
```

Comment:
Why did you move `ENTRYPOINT`?

Reply:
Why are we copying files after the `ENTRYPOINT`?

Reply:
@alexshtin

Reply:
yes, why did you move it from line 24 to 41?

Reply:
Please move it back, and lines 39 to 23.

Reply:
@alexshtin why would we want to copy files after ENTRYPOINT? 
It does not make sense. 

Can you please elaborate on why it is the way it is?